File: pset01/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void create_pyramid(int h);

int main(void)
{
    int h;
    do
    {
        h = get_int("Height: ");
    }
    while (h < 1 || h > 8);
    create_pyramid(h);
}

void create_pyramid(int h)
{
    for (int i = 1; i <= h; i++)
    {
        int hashes = i;
        int spaces = h - i;
        for (int j = 1; j <= spaces; j++)
        {
            printf(" ");
        }
        for (int j = 1; j <= hashes; j++)
        {
            printf("#");
        }

        printf("  ");

        for (int j = 1; j <= i; j++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: pset_6/sentimental-mario-more/mario.py
def main():
    # Demand input between 1 and 8
    while True:
        try:
            h = int(input("Height: "))
        except ValueError:
            continue
        if h >= 1 and h <= 8:
            break

    # Print a half-pyramid made out of hashes
    for i in range(1, h + 1):
        hashes = i
        spaces = h - i

        print(' ' * spaces, end='')
        print('#' * hashes, end='')  # Left half-pyramid

        print('  ', end='')  # Double space between both half-pyramids

        print('#' * hashes, end='')  # Right half-pyramid
        print()  # '\n'


if __name__ == "__main__":
    main()
